lib/tools.js

```diff
@@ -16,9 +16,15 @@ const ipaddr = require('ipaddr.js');
 const ObjectId = require('mongodb').ObjectId;
 const log = require('npmlog');
 const addressparser = require('nodemailer/lib/addressparser');
+const yaml = require('js-yaml');
 
 let templates = false;
 
+if (typeof structuredClone !== 'function') {
+    // eslint-disable-next-line no-global-assign
+    structuredClone = obj => JSON.parse(JSON.stringify(obj));
```

Comment:
All variables must be defined, so this line throws an error:

```
ReferenceError: structuredClone is not defined
    at Object.<anonymous> (/home/runner/work/wildduck/wildduck/lib/tools.js:25:21)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Module.load (node:internal/modules/cjs/loader:1076:32)
    at Function.Module._load (node:internal/modules/cjs/loader:911:12)
    at Module.require (node:internal/modules/cjs/loader:1100:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/home/runner/work/wildduck/wildduck/lib/imap-notifier.js:4:15)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
```
It is probably better to create a local variable under a different name that either gets the existing `structuredClone` (if it is set) or falls back to the JSON handler

```js
const structuredCloneWrapper = typeof structuredClone === 'function' ? structuredClone : JSON.parse(JSON.stringify(obj));
```